### ISSUE TITLE
Various rules refer to `except*` as `except`

### ISSUE THREAD

**Issue by jakkdl** [astral-sh/ruff#14791]
I did a pass on all flake8-bugbear rules upstream to check for `except*` handling (https://github.com/PyCQA/flake8-bugbear/pull/500), and then also checked ruff.. which already handles all of them 🥳 But a bunch of them assume `except` in their message, which could perhaps be confusing to users - but tbh it's mostly a nitpick.

I have not checked any rules other than flake8-bugbear, so I suggest you do a pass checking for any other rules that might be referring to `except`.

```python
try:
    a = 1
except* ValueError:
    a = 2
except* ValueError:
    a = 2

try:
    pass
except* ():
    pass

try:
    pass
except* 1:  # error
    pass

try:
    raise ValueError
except* ValueError:
    raise UserWarning
```

```
$ ruff check --select=B foo.py
foo.py:6:9: B025 try-except block with duplicate exception `ValueError`
foo.py:11:1: B029 Using `except ():` with an empty tuple does not catch anything; add exceptions to handle
foo.py:16:9: B030 `except` handlers should only be exception classes or tuples of exception classes
foo.py:22:5: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
Found 4 errors.
```

expected output

```
$ ruff check --select=B foo.py
foo.py:6:9: B025 try-except* block with duplicate exception `ValueError`
foo.py:11:1: B029 Using `except* ():` with an empty tuple does not catch anything; add exceptions to handle
foo.py:16:9: B030 `except*` handlers should only be exception classes or tuples of exception classes
foo.py:22:5: B904 Within an `except*` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
Found 4 errors.
```

**Comment by dylwil3** [comment]
Thanks for the audit and write-up! This seems reasonable to me to try to fix.

**Comment by smokyabdulrahman** [comment]
I can take this if it's available 👌

**Comment by smokyabdulrahman** [comment]
Draft PR opened.
You can comment anytime if my approach doesn't meet the standard of `astral-sh` or `rust`.

P.S. It's my first rust code base contribution.
